app/javascript/components/bootcamp/SolveExercisePage/Tasks/useTasks.tsx

```diff
@@ -8,38 +8,41 @@ import {
 } from 'react'
 import { SolveExercisePageContext } from '../SolveExercisePageContextWrapper'
 import { type NextExercise, completeSolution } from './completeSolution'
-import type { TaskStore } from '../store/taskStore/taskStore'
 import useAnimationTimelineStore from '../store/animationTimelineStore'
 import { launchConfetti } from './launchConfetti'
 import useTestStore from '../store/testStore'
+import { useLocalStorage } from '@uidotdev/usehooks'
+import useTaskStore from '../store/taskStore/taskStore'
 
 export type FinishLessonModalView =
   | 'initial'
   | 'completedExercise'
   | 'completedLevel'
 
-export function useTasks({
-  areAllTasksCompleted,
-  wasFinishLessonModalShown,
-  setWasFinishLessonModalShown,
-}: Pick<
-  TaskStore,
-  | 'areAllTasksCompleted'
-  | 'wasFinishLessonModalShown'
-  | 'setWasFinishLessonModalShown'
->) {
+export function useTasks() {
   const [isFinishModalOpen, setIsFinishModalOpen] = useState(false)
   const [nextExerciseData, setNextExerciseData] = useState<NextExercise | null>(
     null
   )
+  const [modalsShown, setModalsShown] = useLocalStorage<any[]>(
+    'finish-modals-shown',
```

Comment:
Is there a max length key here? If someone finishes 200 exercises, will this key get too large? Would it be better to have a key per exercise, or put this as part of the existing exercise key?